includes/controllers/pages/class-manage-listings.php

```diff
@@ -105,12 +105,16 @@ public function maybe_add_renew_button( $buttons, $listing_id ) {
 		}
 
 		$is_pending_payment = ( 'pending_payment' === $listing_status );
-		$buttons            = sprintf(
-			'<a class="button wpbdp-button renew-listing" href="%s" %s >%s</a>',
-			$is_pending_payment ? esc_url( $listing->get_payment_url() ) : esc_url( $listing->get_renewal_url() ),
-			'target="_blank" rel="noopener"',
-			$is_pending_payment ? esc_html__( 'Pay Now', 'business-directory-plugin' ) : esc_html__( 'Renew Listing', 'business-directory-plugin' )
-		) . $buttons;
+		$payment_url        = $listing->get_payment_url();
+
+		if ( $payment_url ) {
+			$buttons = sprintf(
+				'<a class="button wpbdp-button renew-listing" href="%s" %s >%s</a>',
+				$is_pending_payment ? esc_url( $payment_url ) : esc_url( $listing->get_renewal_url() ),
```

Comment:
Since we are checking for the payment url, are there cases where the renewal url will be missing when it should be included when the payment is not pending?

Reply:
I did consider this and I assumed the renewal function was also dependent on the Payment object and would faill 100% of the time if no payment was present. 

I checked the renewal function again and tested it's internals. You are right, it uses the $listing object and works regardless. 

I've refactored the conditionals to fix this.

Reply:
This helps, but I don't think we want to show the button if pending and no url. Falling back to the renewal url doesn't really make sense to me if the payment is still pending.

Reply:
This isn't right, I pushed different code. Checking.

Reply:
This is outdated. Please review latest changes.

Reply:
I'm not seeing new changes for this. I think we'll need to add a check like this:
`$show_button = !$is_pending_payment || $payment_url;`

Reply:
That's the approach I took. Please review the code here:

https://github.com/Strategy11/business-directory-plugin/blob/087b91854d6af9d613457a2b8d4d198b1e4778ff/includes/controllers/pages/class-manage-listings.php#L110

Reply:
@banago That's not quite the same thing. Your version will show the Renew listing button, even if the payment is pending. It would better not to show the button if the payment is pending and there's no payment url.

Reply:
@stephywells got it, thanks.

Reply:
Can we do this without adding duplication? Here's what I was thinking:

```suggestion
		
		$is_pending_payment = ( 'pending_payment' === $listing_status );
		$pending_url                 = $listing->get_payment_url();
		$show_button               = ! $is_pending_payment || $payment_url;
		if ( $show_button ) {
		    $buttons            = sprintf(
			    '<a class="button wpbdp-button renew-listing" href="%s" %s >%s</a>',
			    $is_pending_payment ? esc_url( $pending_url ) : esc_url( $listing->get_renewal_url() ),
			    'target="_blank" rel="noopener"',
			    $is_pending_payment ? esc_html__( 'Pay Now', 'business-directory-plugin' ) : esc_html__( 'Renew Listing', 'business-directory-plugin' )
		    ) . $buttons;
		}
```